bdns/dns_test.go

```diff
@@ -422,13 +447,12 @@ func TestRetry(t *testing.T) {
 		if err == errTooManyRequests {
 			t.Errorf("#%d, sent more requests than the test case handles", i)
 		}
-		expectedErr := tc.te.errs[tc.expected-1]
-		if err != expectedErr {
+		expectedErr := tc.expected
+		if (expectedErr == nil && err != nil) ||
+			(expectedErr != nil && err == nil) ||
+			(expectedErr != nil && expectedErr.Error() != err.Error()) {
 			t.Errorf("#%d, error, expected %v, got %v", i, expectedErr, err)
 		}
-		if tc.expected != tc.te.count {
```

Comment:
@jmhodges, I couldn't quite figure out what this was testing or how to replace it after my changes. I assume it's important: What should I be testing here?